src/Engine/Events/EventIR.cpp

```diff
@@ -0,0 +1,1131 @@
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <magic_enum.hpp> // TODO: temporary
+
+#include "Engine/Events/EventIR.h"
+#include "Engine/Events2D.h"
+#include "Engine/Graphics/Level/Decoration.h"
+#include "Engine/Objects/NPC.h"
+#include "Engine/Engine.h"
+#include "Engine/mm7_data.h"
+
+static std::string getVariableSetStr(VariableType type, int value) {
+    if (type >= VAR_MapPersistentVariable_0 && type <= VAR_MapPersistentVariable_74) {
+        return fmt::format("MapVars[{}], {}", type - VAR_MapPersistentVariable_0, value);
+    }
+
+    if (type >= VAR_MapPersistentVariable_75 && type <= VAR_MapPersistentVariable_99) {
```

Comment:
Can we raname it to VAR_MapPersistentDecorVariable_0 .. 25 ?

Reply:
0..24 but yeah, I did it.

Reply:
yes, thanks!